samples/corda/corda-simple-application/clients/src/main/kotlin/com/cordaSimpleApplication/client/AssetTransferManager.kt

```diff
@@ -0,0 +1,705 @@
+                username = "clientUser1",
+                password = "test",
+                rpcPort = config["CORDA_PORT"]!!.toInt())
+            try {
+                val proxy = rpc.proxy
+                val assetClaimStatusBytes = proxy.startFlow(::GetAssetClaimStatusByPledgeId, pledgeId!!, expiryTimeSecs!!)
+                    .returnValue.get()
+                println("assetClaimStatusBytes: ${assetClaimStatusBytes.toString(Charsets.UTF_8)}")
+                println("assetClaimStatusBytes: ${assetClaimStatusBytes.contentToString()}")
+            } catch (e: Exception) {
+                println("Error: ${e.toString()}")
+            } finally {
+                rpc.close()
+            }
+        }
+    }
+}
+
+/**
+ * Fetch Asset Pledge State for Transfer associated with contractId/pledgeId.
+ */
+class GetSimpleAssetPledgeStatusByPledgeIdCommand : CliktCommand(name="get-pledge-status-by-pledge-id", help = "Fetch asset pledge state associated with pledgeId.") {
+    val config by requireObject<Map<String, String>>()
+    val pledgeId: String? by option("-pid", "--pledge-id", help="Pledge Id for Pledge State")
+    val recipientNetworkId: String? by option ("-rnid", "--recipient-network-id", help="Importing network id of pledged asset for asset transfer")
+    override fun run() = runBlocking {
+        if (pledgeId == null) {
+            println("Arguments required: --pledge-id.")
+        } else if (recipientNetworkId == null) {
+            println("Arguments required: --recipient-network-id.")
+        } else {
+            val rpc = NodeRPCConnection(
+                host = config["CORDA_HOST"]!!,
+                username = "clientUser1",
+                password = "test",
+                rpcPort = config["CORDA_PORT"]!!.toInt())
+            try {
+                val proxy = rpc.proxy
+                val assetPledgeStatusBytes = proxy.startFlow(::GetAssetPledgeStatusByPledgeId, pledgeId!!, recipientNetworkId!!)
+                    .returnValue.get()
+                val charset = Charsets.UTF_8
+                println("assetPledgeStatus: ${assetPledgeStatusBytes.toString(charset)}")
+                println("assetPledgeStatus: ${assetPledgeStatusBytes.contentToString()}")
+            } catch (e: Exception) {
+                println("Error: ${e.toString()}")
+            } finally {
+                rpc.close()
+            }
+        }
+    }
+}
+
+private fun getReclaimViewAddress(
+    assetCategory: String,
+    assetType: String,
+    assetIdOrQuantity: String,
+    pledgeId: String,
+    pledgerCert: String,
+    exportNetworkId: String,
+    recipientCert: String,
+    importNetworkId: String,
+    pledgeExpiryTimeSecs: String) : String
+{
+    var viewAddress: String
+    var funcName: String? = null
+    var funcArgs: List<String>? = null
+
+    if (assetCategory.equals("token")) {
+        if (importNetworkId.equals("Corda_Network") || importNetworkId.equals("Corda_Network2")) {
```

Comment:
Why do we need to hardcode network names here?

Reply:
Updated this now @VRamakrishna .